### PR TITLE
Add Altera HAL support for Altera NIOSII MAX 10 devices

### PR DESCRIPTION
**Pull request by ramakrishnapallala**

### CODE REVIEW

**Review by nashif — CHANGES_REQUESTED** [review]
please follow the process for importing external components as described here and adapt the commit message:

https://www.zephyrproject.org/doc/contribute/contribute_guidelines.html#identifying-contribution-origin

and https://www.zephyrproject.org/doc/contribute/contribute_non-apache.html

File: ext/hal/altera/altera_hal/HAL/inc/altera_common.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2017, Intel Corporation
```
Comment (andrewboie) [inline ext/hal/altera/altera_hal/HAL/inc/altera_common.h]: Why are we doing full license boilerplate here, instead of doing SPDX-License-Identifier?

**Review by tyrantlizard77 — CHANGES_REQUESTED** [review]
Do the new UART and TIMER drivers work on the existing CPU design?
I am still uneasy about keeping both the new and old drivers in tree.

File: ext/hal/altera/altera_hal/HAL/inc/altera_common.h (hunk 1 above)
Comment (nashif) [reply ext/hal/altera/altera_hal/HAL/inc/altera_common.h]: this is not an original Zephyr file, we need to keep the original license boilerplate.

Comment (ramakrishnapallala) [reply ext/hal/altera/altera_hal/HAL/inc/altera_common.h]: anas, altera_common.h is added by me for Zephyr adoption of Altera HAL.

But do we need to keep SPDX or Intel license? I thought it's in external folder and thought Intel license header would be good.

Comment (ramakrishnapallala) [reply ext/hal/altera/altera_hal/HAL/inc/altera_common.h]: Andrew, the new ones are working fine. I can remove the old ones in reworked patches.

**Comment by nashif** [comment]
> anas, altera_common.h is added by me for Zephyr adoption of Altera HAL.

then this file needs to be APl licensed and marked as such with SPDX tagging

**Review by rgundi — APPROVED** [review]

**Review by SavinayDharmappa — APPROVED** [review]

**Comment by nashif** [comment]
I still do not see this following http://docs.zephyrproject.org/contribute/contribute_non-apache.html, where is the README?

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by ramakrishnapallala** [comment]
@nashif , i have added the README file. Please have a look.

File: ext/hal/altera/README [hunk 2]
```diff
@@ -0,0 +1,11 @@
+The sources in this directory are imported from Altera
```
Comment (nashif) [inline ext/hal/altera/README]: Please follow the template provided in doc/contribute/code_component_README

**Review by nashif — CHANGES_REQUESTED** [review]

File: ext/hal/altera/README (hunk 2 above)
Comment (ramakrishnapallala) [reply ext/hal/altera/README]: @nashif , i have actually provided info like Origin/Purpose/URL etc.. in the initial commit message itself. Also i have referred README's of other ext/hal/ folders are none of them are having this info README.
The README are simply having info like how they can be downloaded and what they do.

If you want, i can add that info in README as well.

File: ext/hal/altera/README [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+   of different NIOSII based soft IP blocks like UART, I2C, SPI etc...
+   and provides the high level API's for the shim drivers.
+
+Dependencies:
+   Some of the Altera HAL drivers have dependency on the newlibc and
+   it needs to be enabled before enabling the Altera HAL.
+
+   The NIOSII port of Zephyr will be accessing the Altera HAL through
+   shim drivers to enable the different soft IP blocks on Zephyr.
+
```
Comment (nashif) [inline ext/hal/altera/README]: You are missing the license here

**Review by nashif — CHANGES_REQUESTED** [review]

File: ext/hal/altera/README (hunk 3 above)
Comment (nashif) [reply ext/hal/altera/README]: License:
   [use SPDX identifier (https://spdx.org/licenses/), such as BSD-3-Clause]

License Link:
   [link to the project's LICENSE file, e.g., https://github.com/xyz/LICENSE]

Comment (ramakrishnapallala) [reply ext/hal/altera/README]: for License, can i copy the whole License statement here?

also for License link, i can just put the same link as SDK download link, is that OK?

**Comment by nashif** [comment]
recheck

**Comment by nashif** [comment]
the external components in this PR have been approved with majority on both TSC&BOARD, merging for 1.10, action will be taken if there will be any objections.

**Review by nashif — APPROVED** [review]

**Comment by ramakrishnapallala** [comment]
Ok thanks @nashif  :)

**Pull request merged**